File: app/src/main/java/net/devemperor/dictate/usage/UsageModel.java
package net.devemperor.dictate.usage;

public class UsageModel {
    private final String modelName;
    private final long audioTime;
    private final long inputTokens;
    private final long outputTokens;
    private final long modelProvider;

    public UsageModel(String modelName, long audioTime, long inputTokens, long outputTokens, long modelProvider) {
        this.modelName = modelName;
        this.audioTime = audioTime;
        this.inputTokens = inputTokens;
        this.outputTokens = outputTokens;
        this.modelProvider = modelProvider;
    }

    public String getModelName() {
        return modelName;
    }

    public long getAudioTime() {
        return audioTime;
    }

    public long getInputTokens() {
        return inputTokens;
    }

    public long getOutputTokens() {
        return outputTokens;
    }

    public long getModelProvider() {
        return modelProvider;
    }
}

File: app/src/main/kotlin/dev/patrickgold/florisboard/dictate/data/usage/UsageModel.kt
/*
 * Copyright (C) 2026 DevEmperor (Dictate)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 */

package dev.patrickgold.florisboard.dictate.data.usage

/**
 * Accumulated usage for a single model, ported 1:1 from the original Dictate `UsageModel`.
 * Backed by `usage.db` (see [UsageDatabaseHelper] and `docs/COMPATIBILITY.md`).
 */
data class UsageModel(
    val modelName: String,
    val audioTime: Long,
    val inputTokens: Long,
    val outputTokens: Long,
    val provider: Long,
)
